File: client/src/app/enumeration/classificacao.ts
import { SelectItem } from "primeng/components/common/selectitem";

export enum Classificacao {
    INCLUIDA = <any>"Incluída",
    ALTERADA = <any>"Alterada",
    EXCLUIDA = <any>"Excluída"
}

export namespace Classificacao {
    export function getSelectItem(): any {
        var item: SelectItem[] = [];
        item.push({label:'Selecione', value:null});
        item.push({label:Classificacao.INCLUIDA.toString(), value:Classificacao[Classificacao.INCLUIDA]});
        item.push({label:Classificacao.ALTERADA.toString(), value:Classificacao[Classificacao.ALTERADA]});
        item.push({label:Classificacao.EXCLUIDA.toString(), value:Classificacao[Classificacao.EXCLUIDA]});
        return item;
    } 
}

File: server/src/main/java/gmetrica/enumeration/Classificacao.java
package gmetrica.enumeration;

import java.text.Normalizer;

public enum Classificacao {
	
	INCLUIDA ("Incluída"),
	ALTERADA ("Alterada"),
	EXCLUIDA ("Excluída");
	
	private String descricao;
	
	private Classificacao(String descricao) {
		this.descricao = descricao;
	}

	public String getDescricao() {
		return descricao;
	}

	public static Classificacao fromDescricao(String descricao) {
		if(descricao == null) {
			return null;
		}
		
		for(Classificacao valor: Classificacao.values()) {
			
			if(valor.getDescricao().equals(descricao)) {
				return valor;
			}
		}
		
		return null;
	}
	
	public static Classificacao fromContains(String descricao) {
		if(descricao == null) {
			return null;
		}
		
		for(Classificacao valor : Classificacao.values()) {
			descricao = removerAcentos(descricao);
			
			if(removerAcentos(valor.getDescricao()).contains(descricao)) {
				return valor;
			}
		}
		return null;
	}

	public static String removerAcentos(String str) {
	    return Normalizer.normalize(str, Normalizer.Form.NFD).replaceAll("[^\\p{ASCII}]", "").toUpperCase();
	}
}
